src/core/qgismobileapp.cpp

```diff
@@ -553,23 +555,27 @@ void QgisMobileapp::initDeclarative()
   qRegisterMetaType<SnappingResult>( "SnappingResult" );
 
   // Calculate device pixels
-  qreal dpi = mApp->primaryScreen()->logicalDotsPerInch();
-  dpi *= mApp->primaryScreen()->devicePixelRatio();
+  qreal dpi = mApp ? mApp->primaryScreen()->logicalDotsPerInch() * mApp->primaryScreen()->devicePixelRatio() : 96;
 
   // Register some globally available variables
-  rootContext()->setContextProperty( "qVersion", qVersion() );
-  rootContext()->setContextProperty( "ppi", dpi );
-  rootContext()->setContextProperty( "systemFontPointSize", PlatformUtilities::instance()->systemFontPointSize() );
-  rootContext()->setContextProperty( "mouseDoubleClickInterval", QApplication::styleHints()->mouseDoubleClickInterval() );
+  engine->rootContext()->setContextProperty( "qVersion", qVersion() );
+  engine->rootContext()->setContextProperty( "ppi", dpi );
+  engine->rootContext()->setContextProperty( "withNfc", QVariant( NearFieldReader::isSupported() ) );
+  engine->rootContext()->setContextProperty( "systemFontPointSize", PlatformUtilities::instance()->systemFontPointSize() );
+  engine->rootContext()->setContextProperty( "mouseDoubleClickInterval", QApplication::styleHints()->mouseDoubleClickInterval() );
+  engine->rootContext()->setContextProperty( "appVersion", qfield::appVersion );
+  engine->rootContext()->setContextProperty( "appVersionStr", qfield::appVersionStr );
+  engine->rootContext()->setContextProperty( "gitRev", qfield::gitRev );
+  engine->rootContext()->setContextProperty( "platformUtilities", PlatformUtilities::instance() );
+}
+
+void QgisMobileapp::registerGlobalVariables()
```

Comment:
because if we need `initDeclarative` to be static, then we cannot use private members in it.
so to register private members i add this function. maybe rename?